src/network/wait-online/wait-online.c

```diff
@@ -83,14 +83,12 @@ static int parse_interface_with_operstate_range(const char *str) {
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                        "Invalid interface name '%s'", ifname);
 
-        r = hashmap_ensure_allocated(&arg_interfaces, &string_hash_ops);
-        if (r < 0)
+        r = hashmap_ensure_put(&arg_interfaces, &string_hash_ops, ifname, TAKE_PTR(range));
+        if (r == -ENOMEM)
                 return log_oom();
-
-        r = hashmap_put(arg_interfaces, ifname, TAKE_PTR(range));
-        if (r < 0)
+        else if (r < 0)
```

Comment:
```else``` are not really needed, given it returns? just a nit

Reply:
Updated thanks for the review.